backend/sozisel/lib/sozisel_web/schema/resolvers/presenter_resolvers.ex

```diff
@@ -0,0 +1,91 @@
+defmodule SoziselWeb.Schema.Resolvers.PresenterResolvers do
+  alias SoziselWeb.Schema.Helpers
+  alias SoziselWeb.Schema.Subscriptions.Topics
+
+  alias Sozisel.Model.{
+    Sessions,
+    Events,
+    LaunchedEvents
+  }
+
+  alias Events.Event
+  alias LaunchedEvents.LaunchedEvent
+  alias Sessions.Session
+
+  def prepare_data_for_participants(
+        %Event{} = event,
+        %LaunchedEvent{} = launched_event
+      ) do
+    event_data = event.event_data
+
+    participant_event_data = Events.marshal_participant_event_data(event_data)
+
+    %{
+      id: launched_event.id,
+      name: event.name,
+      event_data: participant_event_data
+    }
+  end
+
+  def send_event_to_participants(
+        _parent,
+        %{event_id: event_id, session_id: session_id, broadcast: true},
+        _ctx
+      ) do
+    with %Event{} = event <- Events.get_event(event_id),
```

Comment:
Those two implementations of `send_event_to_participants` look very similar, you could just make a separate function for sending the actual event.
```elixir
def publish_event(%{broadcast: true}) ...
def publish_event(%{broadcast: false, target_participant}) ...
```
If you do so then don't pattern match on broadcast inside `send_event_to_participants`, just in `publish_event`.